### PR TITLE
Lightened turbines

### PR DESCRIPTION
**Pull request by yagalone**
## Summary
Balance "Gas turbine weight reduction according to real life equivalents"

## Purpose of change
The current weight of the turbines makes them a poor choice for most cases, as their power to weight ratio is *worse* than diesel engines and they are far less efficient with fuel, which makes them undesirable for aircraft.
See [this reddit post](https://www.reddit.com/r/cataclysmbn/comments/16pnrse/engines_and_fuel_efficiency_questions/) and PR #3383 for discussions on this topic.

## Describe the solution
Setting the weights of the small, medium and large gas turbines to 150, 200 and 450 kg respectively.

## Describe alternatives you've considered
N/A

## Testing
Loading a game, spawning the 3 kind of turbines and checking their weights.

## Additional context
- The volume of the turbines seems to be quite small, maybe it could be checked as well...
- There might be a use to split the turbines in two categories : Light turbines (fuel guzzlers and relatively flimsy, but with a low weight) and heavy-duty turbines (more efficient and sturdier, but the extra parts for those enhancements makes them heavier).

### CODE REVIEW

File: data/json/items/vehicle/engine.json [hunk 1]
```diff
@@ -307,8 +307,9 @@
     "copy-from": "engine_gasoline",
     "type": "ENGINE",
     "name": { "str": "1350 hp gas turbine engine" },
+    "//": "Weight based on the General Electric CT58-110-1 gas turbine. Used to be widely used on light/medium helicopters, but a bit old for Cataclysm timeframe. If someone have a better fitting equivalent, feel free to update this part.",
     "description": "A gas turbine engine, usually used for military vehicles.  Known for its high rate of fuel consumption, but installing multiple turbines will not affect vehicle power.",
-    "weight": "1130000 g",
+    "weight": "150000 g",
```
Comment (chaosvolt) [inline data/json/items/vehicle/engine.json:312]: Also works as:
```suggestion
    "weight": "150 kg",
```

File: data/json/items/vehicle/engine.json [hunk 2]
```diff
@@ -319,8 +320,9 @@
     "copy-from": "engine_gasoline",
     "type": "ENGINE",
     "name": { "str": "1900 hp gas turbine engine" },
+    "//": "Weight based on the General Electric T700-GE-701D gas turbine, a 1944 hp engine used on several variants of AH-64 Apache and UH-60 Blackhawk.",
     "description": "A large gas turbine engine, usually used for military helicopters.  Known for its high rate of fuel consumption, but installing multiple turbines will not affect vehicle power.",
-    "weight": "1582000 g",
+    "weight": "200000 g",
```
Comment (chaosvolt) [inline data/json/items/vehicle/engine.json:325]:
```suggestion
    "weight": "200 kg",
```

File: data/json/items/vehicle/engine.json [hunk 3]
```diff
@@ -331,8 +333,9 @@
     "copy-from": "engine_gasoline",
     "type": "ENGINE",
     "name": { "str": "6000 hp gas turbine engine" },
+    "//": "Weight based on the Rolls Royce Allison T406-AD-400 gas turbine.",
     "description": "A massive gas turbine engine, used to power the V-22 Osprey.  Known for its high rate of fuel consumption, but installing multiple turbines will not affect vehicle power.",
-    "weight": "3164000 g",
+    "weight": "450000 g",
```
Comment (chaosvolt) [inline data/json/items/vehicle/engine.json:338]:
```suggestion
    "weight": "450 kg",
```

**Comment by chaosvolt** [comment]
Now to hope none of the vehicle efficiency tests use anything with a turbine engine because uuuugh I hate having to figure out how to fix those tests.

File: data/json/items/vehicle/engine.json (hunk 1 above)
Comment (yagalone) [reply data/json/items/vehicle/engine.json:312]: Good idea, it's clearer. But the whole file is rife with such kind of figures...

**Review by chaosvolt — APPROVED** [review]

**Comment by KheirFerrum** [comment]
I feel like these should get a bit more review before merging... Most of these turbines have had their weight lowered by an order of magnitude. I _assume_ that their original weight was mainly to make them unappealing except for larger craft where they don't take up most of the weight. Suddenly reducing them by so much could mess with it by making turbines the best option in every case, which I don't think should be the case.

**Comment by chaosvolt** [comment]
Given they:
1. Require visiting specific locations to acquire, generally military locations for the better ones.
2. Require 7 mechanics skill to install, higher than any of the standard engines.
3. Have WAY higher base fuel consumption, with the small turbine starting off about 4 times that of the V12s.

It generally seems a reasonable tradeoff to me. The first two turbines are lighter than the V8 and V12 engines, so that could be a concern, though again they're just inefficienct in a different way unless you're trying to make a flying vehicle, while the best turbine is still heavier than the V12s.

**Comment by Krwak** [comment]
#1692 wasn't merged due to concerns of developers of helicopters being usable for loot hauling and flying fortresses.
On one hand i'm happy that this is added but on the other rather concerned about the inconsistency here.

**Comment by Krwak** [comment]
> Given they:
> 
>     1. Require visiting specific locations to acquire, generally military locations for the better ones.
> 
>     2. Require 7 mechanics skill to install, higher than any of the standard engines.
> 
>     3. Have WAY higher base fuel consumption, with the small turbine starting off about 4 times that of the V12s.
> 
> 
> It generally seems a reasonable tradeoff to me. The first two turbines are lighter than the V8 and V12 engines, so that could be a concern, though again they're just inefficienct in a different way unless you're trying to make a flying vehicle, while the best turbine is still heavier than the V12s.

Skill and difficulty of looting requirements shouldn't be an argument when considering things that affect game balance and workings of vehicles.

**Comment by scarf005** [comment]
> https://github.com/cataclysmbnteam/Cataclysm-BN/pull/1692 wasn't merged due to concerns of developers that helicopters will become usable for loot hauling and flying fortresses.

i didn't know there existed an earlier PR. i'm sorry for the frustration you had. i think the differences came from the changes of team member and development style. back then, Coolthulhu(BFN) did all the testing and merging, which took long. making changes were hard back then. nowadays, with increased team members to review PRs, the span got much faster, and balances changes are getting merged more quickly.

> While they envision them (helicopters) only as personal carriers. At least that's what i got in discord chat.

i think using helicopters as flying fortress is a great idea. at least majority of korean players do.

> On one hand i'm happy that this is added but on the other rather concerned about the inconsistency here.

please feel free to open a balance PR. as always, thanks for contributing to the BN ecosystem.

**Pull request merged**